storage/series/reader.go

```diff
@@ -179,11 +177,17 @@ func (r Reader) fetchBlocksWithBlocksMapAndBuffer(
 	seriesBlocks block.DatabaseSeriesBlocks,
 	seriesBuffer databaseBuffer,
 ) ([]block.FetchBlockResult, error) {
-	res := make([]block.FetchBlockResult, 0, len(starts))
-
-	var clonedID ts.ID
-	cachePolicy := r.opts.CachePolicy()
-
+	var (
+		// TODO(r): pool these results arrays
+		res         = make([]block.FetchBlockResult, 0, len(starts))
+		cachePolicy = r.opts.CachePolicy()
+		// NB(r): Always use nil for OnRetrieveBlock so we don't cache the
+		// series after fetching it from disk, the fetch blocks API is called
+		// during streaming so to cache it in memory would mean we would
+		// eventually cache all series in memory when we stream results to a
+		// peer.
+		onRetrieve block.OnRetrieveBlock
```

Comment:
This makes sense to me, its just a little confusing because its not clear that a method called FetchBlocks would only be used during peer-bootstrapping. Not sure how to improve it though